packages/styled-components/src/utils/stylis.js

```diff
@@ -0,0 +1,100 @@
+import Stylis from 'stylis/stylis.min';
+import _insertRulePlugin from 'stylis-rule-sheet';
+import { EMPTY_ARRAY, EMPTY_OBJECT } from './empties';
+
+const COMMENT_REGEX = /^\s*\/\/.*$/gm;
+
+export type Stringifier = (
+  css: string,
+  selector: string,
+  prefix: ?string,
+  componentId: string
+) => Array<string>;
+
+export default function createStylisInstance(
+  options: Object = EMPTY_OBJECT,
+  plugins: Array<Function> = EMPTY_ARRAY
+) {
+  const stylis = new Stylis({
+    global: false,
+    cascade: true,
+    keyframe: false,
+    prefix: true,
+    compress: false,
+    semicolon: false, // NOTE: This means "autocomplete missing semicolons"
+
+    // user-given overrides
+    ...options,
+  });
+
+  // Wrap `insertRulePlugin to build a list of rules,
+  // and then make our own plugin to return the rules. This
+  // makes it easier to hook into the existing SSR architecture
+
+  let parsingRules = [];
+
+  // eslint-disable-next-line consistent-return
+  const returnRulesPlugin = context => {
+    if (context === -2) {
```

Comment:
What does `-2` stand for here? Could this maybe made clearer with an explicit mapping / var naming?

Reply:
Honestly, I'm not sure. The docs for stylis are kind of bad so it was trial and error to get it working. This isn't new code, just moved from a different file.

Reply:
It's documented here https://github.com/thysultan/stylis.js#plugins

We might want to extract this to some named constants. `-2` is the last step in the plugin lifecycle, so it allows us to return a different result from the stylis function itself after it's done processing all CSS rules.

Reply:
Interesting. I imagine this happens often enough that looking this up on an object will incur a perf hit?

Reply:
Maybe just a comment linking to that section here would be enough?